midterm-2023-11-06/code/5088-darko-petrevski/script.js

```diff
@@ -0,0 +1,126 @@
+document.addEventListener('DOMContentLoaded', () => {
+    const bandsTable = document.getElementById('bandsTable');
+    const nameFilter = document.getElementById('nameFilter');
+    const countryFilter = document.getElementById('countryFilter');
+    const genreFilter = document.getElementById('genreFilter');
+    const yearActiveFilter = document.getElementById('yearActiveFilter');
+
+    let bandsData = []; 
+
+    fetch('https://raw.githubusercontent.com/sweko/uacs-internet-programming-exams/main/midterm-2023-11-06/data/bands.json')
+        .then(response => response.json())
+        .then(data => {
+            bandsData = data.metalBands;
+            populateTable(bandsData);
+            populateFilterDropdowns(bandsData);
+        })
+        .catch(error => console.error('Error fetching data:', error));
+
+    function populateTable(data) {
+        const table = document.getElementById('bandsTable');
+    
+        table.innerHTML = '';
+    
+        const headers = [
+            'ID',
+            'Name',
+            'Formed',
+            'Location',
+            'Country',
+            'Genre',
+            'Members',
+            'Albums',
+            'First Album',
+            'Last Album',
+            'Years Active'
+        ];
+    
+        const headerRow = table.insertRow();
```

Comment:
We don't need to generate the header each time, as the data always has the same format. If we're using a table, we can use the `thead` and `tbody` elements to actually only replace the body.